crates/pop-cli/src/commands/build/spec.rs

```diff
@@ -127,157 +123,327 @@ pub(crate) enum RelayChain {
+			genesis_state,
+			genesis_code,
+		} = self;
 
-		// Create output path if needed
-		let mut output_path = self.output_file.unwrap_or_else(|| PathBuf::from("./"));
-		let mut plain_chain_spec;
-		if output_path.is_dir() {
-			if !output_path.exists() {
-				// Generate the output path if needed
-				create_dir_all(&output_path)?;
+		// Chain.
+		let chain = match chain {
+			Some(chain) => chain,
+			_ => {
+				// Prompt for chain if not provided.
+				input("Specify the chain specification. It can be one of the predefined ones (e.g. dev, local or a custom one) or the path to an existing chain spec.")
+					.placeholder("dev")
+					.default_input("dev")
+					.interact()?
+			},
+		};
+
+		// Output file.
+		let maybe_chain_spec_file = PathBuf::from(&chain);
+		// Check if the provided chain specification is a file.
+		let (output_file, prompt) = if maybe_chain_spec_file.exists() &&
+			maybe_chain_spec_file.is_file()
+		{
+			if output_file.is_some() {
+				cli.warning("NOTE: If an existing chain spec file is provided it will be used for the output path.")?;
 			}
-			plain_chain_spec = output_path.join(DEFAULT_SPEC_NAME);
+			// Prompt whether the user wants to make additional changes to the provided chain spec
+			// file.
+			let prompt = confirm("An existing chain spec file is provided. Do you want to make additional changes to it?".to_string())
+				.initial_value(false)
+				.interact()?;
+			// Set the provided chain specification file as output file and whether to prompt the
+			// user for additional changes to the provided spec.
+			(maybe_chain_spec_file, prompt)
 		} else {
-			plain_chain_spec = output_path.clone();
-			output_path.pop();
-			if !output_path.exists() {
-				// Generate the output path if needed
-				create_dir_all(&output_path)?;
-			}
-		}
-		plain_chain_spec.set_extension("json");
+			let output_file = match output_file {
+				Some(output) => output,
+				None => {
+					// Prompt for output file if not provided.
+					let default_output = format!("./{DEFAULT_SPEC_NAME}");
+					input("Name of the plain chain spec file. If a path is given, the necessary directories will be created:")
+						.placeholder(&default_output)
+						.default_input(&default_output)
+						.interact()?
+				},
+			};
+			(prepare_output_path(output_file)?, true)
+		};
+		// If chain specification file already exists, obtain values for defaults when prompting.
+		let chain_spec = ChainSpec::from(&output_file).ok();
 
-		// Locate binary, if it doesn't exist trigger build.
-		let mode: Profile = self.release.into();
-		let cwd = current_dir().unwrap_or(PathBuf::from("./"));
-		let binary_path = match binary_path(&mode.target_directory(&cwd), &cwd.join("node")) {
-			Ok(binary_path) => binary_path,
+		// Para id.
+		let id = match id.or_else(|| {
+			chain_spec.as_ref().and_then(|cs| cs.get_parachain_id().map(|id| id as u32))
+		}) {
+			Some(id) if !prompt => id,
+			// No para id provided or user wants to make changes.
 			_ => {
```

Comment:
If user provides `id` as flag (or any of the other `Option<>` fields in the `BuildSpecCmd`) and does not want to make a change to the provided chain spec file (the user is prompted whether it wants to make changes to the file), the flag’s value is used for the chain spec file. If it also wants to make changes to the provided chain spec file, it will see the provided `id` flag value as default but is still able choose another number. While the flow might be a complicated it allows for anything and it felt clear to me as a user.